app/src/lib/git/for-each-ref.ts

```diff
@@ -74,10 +74,34 @@ export async function getBranches(
       continue
     }
 
+    const existsOnRemote =
+      upstream !== ''
+        ? await checkBranchExistsOnRemote(upstream, repository)
```

Comment:
Some repositories can have a ton of branches, so performing an operation on each one isn't reasonable. We need to either get this from `for-each-ref` (though in a cursory look I didn't see a way, maybe with `symref`?) or we need to do this check only when needed.

Reply:
Seconding what @joshaber says here about this scaling to hundreds of refs. Could we just defer this to when you delete the branch, as that will probably be a bit more accurate?